ros_cross_compile/ros_cross_compile.py

```diff
@@ -174,7 +181,15 @@ def cross_compile_pipeline(
 def main():
     """Start the cross-compilation workflow."""
     args = parse_args(sys.argv[1:])
-    cross_compile_pipeline(args)
+    ros_workspace_dir = resolve_ros_workspace(args.ros_workspace)
+    data_collector = DataCollector()
+    data_writer = DataWriter(ros_workspace_dir)
+
+    try:
+        with data_collector.data_timer('e2e'):
```

Comment:
descriptive names please :) "cross_compile_pipeline_full" may feel clunky at first but nobody will ever be confused about what it is